src/hardware/gus.cpp

```diff
@@ -1552,30 +1566,8 @@ void Gus::WriteToRegister()
 		}
 		return;
 	case 0x4c: // Reset register
-	{
-		// The reset register bits control runtime state such as
-		// stoping, starting, and if the DAC or IRQs will be used.
-
-		const auto reset_register = static_cast<uint8_t>(register_data >> 8);
-
-		// From section 2.6.1.9, page 16, of the UltraSound Software
-		// Development Kit (SDK) Version 2.22
-
-		// Bit 0 - GF1 Master Reset. 0 = reset, 1 = run. As long as this
-		//         is a 0, it will be held in a reset state.
-		if (bit::cleared(reset_register, bit::literals::b0)) {
-			StopPlayback();
-		} else {
-			BeginPlayback();
-		}
-
-		// Bit 1 - Enable DAC output. DAC's will not run unless its set.
-		dac_enabled = bit::is(reset_register, bit::literals::b1);
-
-		// Bit 2 - Master IRQ enable. This bit MUST be set to get ANY of
-		//         the GF1-generated IRQs (wavetable, volume, etc).
-		irq_enabled = bit::is(reset_register, bit::literals::b2);
-	}
+		reset_register.data = static_cast<uint8_t>(register_data >> 8);
```

Comment:
I think you can simply just do this, and that's preferable:

`reset_register = {register_data >> 8};`

That's how I'm doing it in `void SurroundProcessor::ControlWrite(const uint8_t val)` in `adlib_gold.cpp`.

Reply:
Thanks for trying anyway 😄